### PR TITLE
9 seperate topics page

### PR DESCRIPTION
**Pull request by moshkh**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -23,10 +22,15 @@ function App() {
         <UserBar />
         <Routes>
           <Route path="/" element={<Intro />} />
+          <Route path="/articles/allarticles" element={<Articles />} />
```
Comment (callumu21) [inline src/App.js]: These are some strange paths. Ideally, you should have:

1. /articles
2. /articles/:article_id
3. /topics
4. /topics/:topic

File: src/Components/Articles.jsx [hunk 2]
```diff
@@ -1,33 +1,47 @@
...
   useEffect(() => {
-    getArticles().then((res) => {
-      setArticles(res);
-      setLoading(false);
-    });
-  }, []);
+    getArticles()
+      .then((articlesReceived) => {
+        if (topicname) {
+          const filteredArticles = articlesReceived.filter((article) => {
```
Comment (callumu21) [inline src/Components/Articles.jsx]: You shouldn't be filtering here. Retrieving the filtered list by topic name should be handled by your API request

**Review by callumu21 — CHANGES_REQUESTED** [review]
Good start, Musab! Just a couple of changes needed:

1. Retrieving the filtered list of articles should be handled by your API call, not by using .filter on all articles yourself. Have a look at your back end and accepted queries if you need a reminder.
2. Your routes are a little strange and should be refactored to make them more intuitive. See my comment for more instructions.

Once you've incorporated these, please resubmit this PR 👍

File: src/App.js (hunk 1 above)
Comment (moshkh) [reply src/App.js]: I forgot about the my BE /api/articles allowing a query for topic - so I was doing all sorts of code gymnastics to get it to work! which including weird routes

File: src/Components/Articles.jsx [hunk 3]
```diff
@@ -1,28 +1,34 @@
...
-      setLoading(false);
-    });
-  }, []);
+    getArticles(topicname)
+      .then((res) => {
+        setArticles(res);
+        setLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
```
Comment (callumu21) [inline src/Components/Articles.jsx:19]: You should be doing something here for the user, other than logging the error. If your get request fails, you should indicate that through some text on screen. Not required for this ticket, but something to consider when it comes to error handling in ticket 12

File: src/App.js [hunk 4]
```diff
@@ -23,10 +22,14 @@ function App() {
         <UserBar />
         <Routes>
           <Route path="/" element={<Intro />} />
-          <Route path="/:user" element={<CurrentUser />} />
           <Route path="/articles" element={<Articles />} />
-          <Route path="/articles/:article_id" element={<SingleArticle currUser={currUser}/>} />
           <Route path="/topics" element={<Topics />} />
+          <Route path="/:user" element={<CurrentUser />} />
+          <Route path="/topics/:topicname" element={<Articles />} />
```
Comment (callumu21) [inline src/App.js:28]: Much better!

**Review by callumu21 — APPROVED** [review]
Nice work! Good to merge 👍

**Pull request merged**